indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/IndyFacade.kt

```diff
@@ -0,0 +1,193 @@
+     */
+    fun issueCredentialAndUpdateLedger(
+        credentialRequest: CredentialRequestInfo,
+        offer: CredentialOffer,
+        revocationRegistryId: RevocationRegistryDefinitionId?,
+        proposalProvider: () -> CredentialProposal
+    ): CredentialInfo
+
+    /**
+     * Stores [Credential] in prover's wallet gathering data using [LedgerService]
+     *
+     * @param credentialInfo [CredentialInfo] - credential and all reliable data
+     * @param credentialRequest [CredentialRequestInfo] - credential request and all reliable data
+     * @param offer [CredentialOffer]
+     */
+    fun receiveCredential(
+        credentialInfo: CredentialInfo,
+        credentialRequest: CredentialRequestInfo,
+        offer: CredentialOffer
+    )
+
+    /**
+     * Revokes previously issued [Credential] using [WalletService] and [LedgerService]
+     *
+     * @param revocationRegistryId [RevocationRegistryDefinitionId] - revocation registry definition id
+     * @param credentialRevocationId [String] - revocation registry credential index
+     */
+    fun revokeCredentialAndUpdateLedger(
+        revocationRegistryId: RevocationRegistryDefinitionId,
+        credentialRevocationId: String
+    ): RevocationRegistryEntry
+
+    /**
+     * Creates [ProofRequest]. This function has nothing to do with Indy API, it is used just to produce well-shaped data.
+     *
+     * @param version [String] - ???
+     * @param name [String] - name of this proof request
+     * @param attributes [List] of [CredentialFieldReference] - attributes which prover needs to reveal
+     * @param predicates [List] of [CredentialPredicate] - predicates which prover should answer
+     * @param nonRevoked [Interval] or [null] - time interval of [attributes] and [predicates] non-revocation
+     * @param nonce [String] - random value to distinct identical proofs
+     *
+     * @return [ProofRequest]
+     */
+    fun createProofRequest(
+        version: String,
+        name: String,
+        attributes: List<CredentialFieldReference>,
+        predicates: List<CredentialPredicate>,
+        nonRevoked: Interval?,
+        nonce: String = "123123"
+    ): ProofRequest
+
+    /**
+     * Creates [ProofInfo] for provided [ProofRequest].
+     *
+     * @param proofRequest [ProofRequest] - proof request created by verifier
+     * @param masterSecretId [String]
+     *
+     * @return [ProofInfo] - proof and all reliable data
+     */
+    fun createProofFromLedgerData(
+        proofRequest: ProofRequest,
+        masterSecretId: String = DEFAULT_MASTER_SECRET_ID
+    ): ProofInfo
+
+    /**
+     * Verifies [ProofInfo] produced by prover
+     *
+     * @param proofReq [ProofRequest] - proof request used by prover to create proof
+     * @param proof [ProofInfo] - proof created by prover
+     *
+     * @return [Boolean] - is proof valid?
+     */
+    fun verifyProofWithLedgerData(proofReq: ProofRequest, proof: ProofInfo): Boolean
```

Comment:
May be naming should not contain WithLedger\UpdateLedger?
If use case assumes that there is no other way, then why we need to clarify?

Reply:
Idk. I thought it would be much cleaner with these annotations what does this method actually do.